### DISCUSSION TITLE
ArgoCD gitops from UI

### DISCUSSION THREAD

**Discussion by aperullo** [argoproj/argo-cd#7185] [Q&A]
Hello, I just set up Argocd for the first time to play around with and I'm unable to figure out one thing. How is the gitops is supposed to work from the UI?

I create an app in the UI and then an Application CRD shows up in the cluster, but where's the application.yaml and git repository that the UI committed those to? Having the CR show up is good but I don't quite have the visibility I'd expect of being able to see cluster state through a control repo. I can't find any indication in the docs either.

I have to imagine the UI is involved with git somewhere but I haven't configured it with a git repository anywhere. Does the control-repo live in the argo-repo-server or something?

How do I find out/how do I do this?

**Comment by wanghong230** [comment]
Application part is not Git tracked, but metadata as CR in the control plane cluster. If you use app of apps pattern, then Yes. You are creating other applications with one application, so applications YAMl will be in the Git.

**Reply by aperullo** [reply]
But its not really a closed gitops loop then if the UI creates and runs applications without putting that configuration in git. That allows imperative deployment that could be lost if the cluster goes down doesn't it?

Is there a way to get the CRDs that the UI creates to be committed somewhere so that it can persist?

I would think the full gitops loop would be:
1. Argo UI writes application CRs into yaml that go to a control repo.
2. Argo-repo-server reads the control-repo and applies them into the cluster (Like what the Flux 2 Source and Kustomize Controllers do)
3. Argo-server reads the CRs and actually deploys them. 

I don't want my developers to have to define application CRs just to be able to deploy, I want them to have a nice UI. But at the same time I want that UI to place things in git so they are persisted.

**Reply by wtam2018** [reply]
Argo CD is not opinionated about requiring Application CRs to be saved in a control repository.   It takes an simple approach to sync the source of truth in git to the cluster without writing anything back to the git repo.   (Think about it ... you can add hundreds or thousands of application CRs to be manged in your repo if you have that many applications to deploy.   I don't know if you want to mandate it.)  You can certainly merge Application CRs to a repo and even have Argo CD to deploy it which is the [app-of-apps pattern ](https://argoproj.github.io/argo-cd/operator-manual/cluster-bootstrapping/) Hong mentions.   The ApplicationSet also provides some powerful way to generate/deploy applications https://argocd-applicationset.readthedocs.io/en/stable/Use-Cases/ that you may want to take a look.
